optimism/src/mips/witness.rs

```diff
@@ -654,9 +654,12 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
             }
             self.keccak_env = Some(keccak_env);
         }
-        // Reset Keccak environment
-        self.preimage_bytes_read = Some(0);
+        // Reset environment
+        self.preimage_bytes_read = None;
+        self.preimage = None;
+        self.preimage_key = None;
         self.hash_count += 1;
```

Comment:
hash count should be increase in the previous scope, when we read the whole preimage.